noir_stdlib/src/hash.nr

```diff
@@ -46,8 +51,34 @@ pub fn pedersen_hash<N>(input: [Field; N]) -> Field
     pedersen_hash_with_separator(input, 0)
 }
 
-#[foreign(pedersen_hash)]
-pub fn pedersen_hash_with_separator<N>(input: [Field; N], separator: u32) -> Field {}
+#[field(bn254)]
+fn derive_generators<N, M>(
+    domain_separator_bytes: [u8; M],
+    num_generators: u32,
+    starting_index: u32
+) -> [EmbeddedCurvePoint; N] {
+    crate::assert_constant(domain_separator_bytes);
+    crate::assert_constant(num_generators);
+    crate::assert_constant(starting_index);
+    __derive_generators(domain_separator_bytes, num_generators, starting_index)
+}
+
+#[builtin(derive_pedersen_generators)]
+#[field(bn254)]
+fn __derive_generators<N, M>(
+    domain_separator_bytes: [u8; M],
+    num_generators: u32,
+    starting_index: u32
+) -> [EmbeddedCurvePoint; N] {}
```

Comment:
Not sure how easy it would be to pull values of the generics from inside of the compiler (so we can remove `num_generators` from here too)

Reply:
Done, I assumed the result is a fixed array.